libraries/rush-lib/src/cli/actions/DeployAction.ts

```diff
@@ -82,19 +93,92 @@ export class DeployAction extends BaseRushAction {
   }
 
   protected async runAsync(): Promise<void> {
-    const deployManagerModule: typeof deployManagerType = await import(
-      /* webpackChunkName: 'DeployManager' */
-      '../../logic/deploy/DeployManager'
+    const scenarioName: string | undefined = this._scenario.value;
+    const scenarioFilePath: string = DeployScenarioConfiguration.getConfigFilePath(
```

Comment:
Consider async importing `DeployScenarioConfiguration` as well? Or just adopt the model of having the `runAsync()` function async import a separate module that is the actual executor and pass over whatever parameters to it.